hooks/tk-multi-publish2/basic/collector.py

```diff
@@ -188,6 +185,45 @@ def process_current_session(self, settings, parent_item):
         # reset the original document to restore the state for the user
         engine.adobe.app.activeDocument = current_document
 
+    def _set_item_thumbnail_for_document(self, document_item, document):
+        """
+        Creates thumbnail for photoshop document.
+        Input:
+        document_item: Item associated to photoshop document
+        document: Photoshop document.
+        """
+
+        publisher = self.parent
+        engine = publisher.engine
+        path = _document_path(document)
+        if path:
+            # attempt to create thumbnail
+            document_item.set_thumbnail_from_path(path)
+            # if document is .psd, the thumbnail is None
+            if document_item.thumbnail is None:
+                # remember current document
+                current_document_cache = engine.adobe.get_active_document()
+                engine.adobe.app.activeDocument = document
+
+                # create a temporary file to save document as .jpg
+                pathTemp = tempfile.NamedTemporaryFile(
+                           suffix=".jpg",
+                           prefix="sgtk_thumb_temp",
+                           delete=False
+                           ).name
+
+                # save photoshop document as jpg (jpeg is supported by QPixmap).
+                # the "as copy" flag is set to true so that the temporary file won't
+                # be considererd officially as the saved version of the document to publish
+                # or to be edited by photoshop
+                document.saveAs(engine.adobe.File(pathTemp),engine.adobe.JPEGSaveOptions,True)
```

Comment:
Can we use the engine's `save_to_path` here?

Reply:
No we can't. We absolutely need to set the last parameter to True here. save_to_path does not.